Sources/Server/server.swift

```diff
@@ -1 +1,68 @@
+import Foundation
+import Socket
+import Requests
 
+public class Server {
+
+    var listeningSocket: Socket!
+    var clientSocket: Socket!
+    var readData = Data()
+    var parser: RequestParser
+
+    public init(parser: RequestParser) {
+        self.parser = parser
+    }
+
+    public func run() {
+        do {
+            try self.listeningSocket = Socket.create()
+            try self.listeningSocket.listen(on: 5000)
+            repeat {
+                try clientSocket = self.listeningSocket.acceptClientConnection()
+                let parsedIncomingRequest = parseRequest(socket: clientSocket)
+                let httpMethod = parsedIncomingRequest.returnMethod()
+
+                if (httpMethod == "GET" || httpMethod == "POST") {
+                    response200(socket: clientSocket)
+                } else {
+                    response404(socket: clientSocket)
+                }
+            } while true
+        } catch let error {
+            print (error.localizedDescription)
+        }
+    }
+
+    private func response200(socket: Socket) {
+        do {
+            let response = Data("HTTP/1.1 200 OK\r\nContent-Length: 0\r\n\r\n".utf8)
+            try socket.write(from: response)
+        }
+        catch let error {
+            print (error.localizedDescription)
+        }
+    }
+
+    private func response404(socket: Socket) {
+        do {
+            let response = Data("HTTP/1.1 404 NotFound\r\nContent-Length: 0\r\n\r\n".utf8)
```

Comment:
You might want to define a constant CRLF = "\r\n" since it pops up so often in your code

Reply:
➕ I agree - it seems that there may be a few stings that would good to pull out in constant as you continue developing.